server/handlers/medco_node.go

```diff
@@ -66,18 +68,90 @@ func MedCoNodeExploreQueryHandler(params medco_node.ExploreQueryParams, principa
 	for timerName, timerDuration := range query.Result.Timers {
 		milliseconds := int64(timerDuration / time.Millisecond)
 		timers = append(timers, &models.ExploreQueryResultElementTimersItems0{
-			Name: timerName,
+			Name:         timerName,
 			Milliseconds: &milliseconds,
 		})
 	}
 
 	return medco_node.NewExploreQueryOK().WithPayload(&medco_node.ExploreQueryOKBody{
-		ID:     query.ID,
-		Query:  params.QueryRequest.Query,
+		ID:    query.ID,
+		Query: params.QueryRequest.Query,
 		Result: &models.ExploreQueryResultElement{
-			EncryptedCount: query.Result.EncCount,
+			EncryptedCount:       query.Result.EncCount,
 			EncryptedPatientList: query.Result.EncPatientList,
-			Timers: timers,
-			Status: models.ExploreQueryResultElementStatusAvailable,
-	}})
+			Timers:               timers,
+			Status:               models.ExploreQueryResultElementStatusAvailable,
+		}})
+}
+
+// MedCoNodeGetStatusHandler handles /medco/node/status API endpoint
+func MedCoNodeGetStatusHandler(params medco_node.GetStatusParams, principal *models.User) middleware.Responder {
+
+	message := ""
+
+	//test i2b2
+	okI2b2 := testI2b2()
+	if !okI2b2 {
+		message += "i2b2 error:"
+	}
+	//test unlynx
+	okUnlynx := testUnlynx()
+	if !okUnlynx {
+		message += "Unlynx error:"
+	}
+	//test genomic-annotations
+	okGenomicAnnotations := testGenomicAnnotations()
+	if !okGenomicAnnotations {
+		message += "Genomic annotations error"
+	}
+
+	return medco_node.NewGetStatusOK().WithPayload(&medco_node.GetStatusOKBody{
+		Message:  message,
+		StatusOK: okI2b2 && okUnlynx && okGenomicAnnotations,
+	})
+
+	return nil
+
+}
+
+func testI2b2() bool {
+
+	return true
+
+}
+
+func testUnlynx() bool {
+
+	return true
+
+}
+
+func testGenomicAnnotations() bool {
```

Comment:
Those "testX()" functions should be moved to a separate package rather than the handler. Handler should be just the bridging between HTTP API and the actual implementation. E.g. `medco-connector/server/status` package would make sense IMO.

Reply:
Moving those functions to a separate package is a bit excessive IMO, since I do not think they are going to be reused anywhere else. I encapsulated those pieces of code in private functions just for the sake of readability, but I could just copy and paste them into the handler body. What do you think?

Reply:
It's more about them not being in the handlers package: handlers should be the bridging between HTTP API and implementation code. It could be in medcoserver package (medco-connecter/server) rather than its own package, if you think that's better.